src/components/Select/Select.js

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled, { css, cx } from 'react-emotion';
+
+import { textMega } from '../../styles/style-helpers';
+import IconInputWrapper from '../IconInputWrapper';
+import DownTriangleIcon from './down-triangle.svg';
+
+const baseSelectStyles = ({ theme }) => css`
+  label: select;
+  appearance: none;
+  background-color: ${theme.colors.white};
+  border-width: 1px;
+  border-style: solid;
+  border-color: ${theme.colors.n300};
+  border-radius: ${theme.borderRadius.mega};
+  box-shadow: inset 0 1px 2px 0 rgba(102, 113, 123, 0.21);
+  color: ${theme.colors.n900};
+  padding: ${theme.spacings.byte} ${theme.spacings.tera} ${theme.spacings.byte}
+    ${theme.spacings.kilo};
+  position: relative;
+  ${textMega({ theme })};
+  z-index: 30;
+
+  &:focus,
+  &:hover,
+  &:active {
+    outline: none;
+  }
+
+  &:focus {
+    border: 1px solid ${theme.colors.b500};
+  }
+
+  &::after {
+  }
+
+  &:-moz-focusring {
+    color: transparent;
+    text-shadow: 0 0 0 #000;
+  }
+`;
+
+const disabledSelectStyles = ({ disabled }) =>
+  disabled &&
+  css`
+    label: select--disabled;
+    opacity: 0.4;
+    pointer-events: none;
+  `;
+
+const baseIconStyles = ({ theme }) => css`
+  label: select__icon;
+  fill: ${theme.colors.n700};
+  display: block;
+  z-index: 40;
+  pointer-events: none;
+`;
+
+const SelectElement = styled('select', { label: 'SelectElement' })(
+  baseSelectStyles,
+  disabledSelectStyles
+);
+const Icon = styled(DownTriangleIcon, { label: 'SelectIcon' })(baseIconStyles);
+
+/**
+ * A native select component.
+ */
+const Select = ({
+  options,
+  value,
+  placeholder,
+  selector,
+  disabled,
+  ...props
+}) => (
+  <IconInputWrapper
```

Comment:
I noticed the use of this component around. We may turn it into a HOC in the future. 

```js
function IconWrapper(position, Icon, WrappedComponent) {
  return class extends React.Component {
    render() {
      return  <IconInputWrapper
                 {...{ selector }}
                 iconPosition={position}
                 icon={Icon}
                 input={(props) => (<WrappedComponent {...props} />)} />;
    }
  }
}
```
And then:
```javascript
const MyInput = () => <input .... />;

export default IconWrapper("left", <MyIcon />, MyInput);
```

It makes much cleaner. I didn't go through the edge cases, but I think it may be an option. WDYT?

Reply:
Hmmmm. If I understand your correctly, then this would create a HOC that renders a component, which takes render props. I don't think this would be helpful. Are you trying to minimize the "boilerplate" for using this component when, for example, creating an input with an icon? If you want to reduce that, I guess you could turn `IconInputWrapper` itself into a HOC that takes two components. But that's what I didn't do on purpose.

Reply:
Yes. I just saw this boilerplate around and how a little "complex" is its use. Would be nice thinking in a more simple abstraction since it just injects some context to the children